projects/core/src/lib/mask.ts

```diff
@@ -133,9 +134,24 @@ export class Maskito {
         }
     }
 
-    private updateSelectionRange([from, to]: [from: number, to: number]): void {
+    private updateSelectionRange([from, to]: SelectionRange): void {
         if (this.element.selectionStart !== from || this.element.selectionEnd !== to) {
             this.element.setSelectionRange(from, to);
         }
     }
 }
+
+function extendToNotEmptyRange(
+    [from, to]: SelectionRange,
+    isForward: boolean,
+): SelectionRange {
+    if (from !== to) {
+        return [from, to];
+    }
+
+    if (isForward) {
```

Comment:
Could be ternary. And also what do you think if we try to keep one function to one file? Might be easier to support/review.